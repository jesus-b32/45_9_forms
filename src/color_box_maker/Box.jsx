import React from "react";

/**
 * this component should display a div with a background color, width and height based on the props passed to it.
 * @param {Object} {color, width, height} 
 * @returns box component
 */
function Box ({id, backgroundColor, width, height, handleRemove}) {
    const boxStyle = {
        backgroundColor,
        width: `${width}em`,
        height: `${height}em`
    }

    // handle removing a box when click event happens
    const removeHandle = () => handleRemove(id);

    return (
        <div>
            <div style={boxStyle}></div>
            <button onClick={removeHandle}>Remove Box</button>
        </div>
    )
}


export default Box;