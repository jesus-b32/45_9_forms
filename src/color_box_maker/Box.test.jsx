import { render, fireEvent } from '@testing-library/react';
import Box from './Box';
import { describe, it, expect, vi } from 'vitest';

describe('Box', () => {
    //smoke test
    it('renders App component', () => {
        render(<Box/>);
    })

    //snapshot test
    it('matches snapshot', () => {
        const {asFragment} = render(<Box/>);
        expect(asFragment()).toMatchSnapshot();
    })

    it("runs the removeHandle function on button click", function() {
        const removeMock = vi.fn();
        const { getByText } = render(<Box handleRemove={removeMock} />);
        const deleteButton = getByText("Remove Box");
        fireEvent.click(deleteButton);
        expect(removeMock).toHaveBeenCalled();
    });
})