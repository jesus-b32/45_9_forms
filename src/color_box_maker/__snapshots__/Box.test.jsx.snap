// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Box > matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div />
    <button>
      Remove Box
    </button>
  </div>
</DocumentFragment>
`;
